lib/modules/manager/bundler/artifacts.spec.ts

```diff
@@ -128,6 +128,32 @@ describe('modules/manager/bundler/artifacts', () => {
     expect(execSnapshots).toMatchSnapshot();
   });
 
+  it('supports conservative mode', async () => {
+    fs.readLocalFile.mockResolvedValueOnce('Current Gemfile.lock');
+    fs.writeLocalFile.mockResolvedValueOnce(null as never);
+    fs.readLocalFile.mockResolvedValueOnce(null as never);
+    const execSnapshots = mockExecAll(exec);
+    git.getRepoStatus.mockResolvedValueOnce({
+      modified: ['Gemfile.lock'],
+    } as StatusResult);
+    fs.readLocalFile.mockResolvedValueOnce('Updated Gemfile.lock' as any);
+    expect(
+      await updateArtifacts({
+        packageFileName: 'Gemfile',
+        updatedDeps: [{ depName: 'foo' }, { depName: 'bar' }],
+        newPackageFileContent: 'Updated Gemfile content',
+        config: {
+          ...config,
+          postUpdateOptions: [
+            ...(config.postUpdateOptions ?? []),
+            'bundlerConservative',
+          ],
+        },
+      })
+    ).toEqual([updatedGemfileLock]);
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
please use `toMatchObject` and pass a partial object with properties which need to match

Reply:
All the other tests in this file use snapshots; seems like a one-off decision to do an object comparison here. It is a slimmer solution though, so I've pushed that change to see if we like it better.